### ISSUE TITLE
Build failure if `cargo` environment variable exists

### ISSUE THREAD

**Issue by Nemo157** [briansmith/ring#963]
```console
> cargo=test cargo build
    Updating crates.io index
   Compiling libc v0.2.69
   Compiling cc v1.0.50
   Compiling lazy_static v1.4.0
   Compiling untrusted v0.7.0
   Compiling spin v0.5.2
   Compiling ring v0.17.0-alpha.1 (/tmp/tmp.bJ19ZyaNOk/ring-0.16.12/ring-0.17.0-alpha.1)
error: Wrong output in build script of `ring v0.17.0-alpha.1 (/tmp/tmp.bJ19ZyaNOk/ring-0.16.12/ring-0.17.0-alpha.1)`: `cargo: test`
```

**Comment by pietro** [comment]
What's the output of `cargo=test cargo build -vv`?

**Comment by Nemo157** [comment]
The bug is in this loop, it will print `cargo: test` and presumably cargo attempts to parse any lines starting with `cargo.*:`.

https://github.com/briansmith/ring/blob/00c21e253ba9cd3b66ab41155414b0d0e91b6c95/build.rs#L263-L265

**Comment by HeapUnderfl0w** [comment]
i can confirm this: https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script

**Comment by briansmith** [comment]
:sigh: Thanks for pointing this out. I suggest we change the line from:
```
     println!("{}: {}", key, value); 
```
to
```
     println!("ENV {}={}", key, value); 
```

**Comment by briansmith** [comment]
Also, maybe all this output should be `eprintln!` and/or `log!` output.

**Comment by briansmith** [comment]
Fixed in #968.
